src/tiled/automapper.cpp

```diff
@@ -749,7 +797,7 @@ static bool compareLayerTo(const TileLayer *setLayer,
                 if (listNo.isEmpty()) {
                     if (matchListYes)
                         continue;
-                    if (!ruleDefinedListYes && !cells.contains(c1))
```

Comment:
@axboureau I agree the system should work more intuitively, but do you have a specific suggestion of what alternative to offer for people relying on this behavior?